numpy/linalg/umath_linalg.c.src

```diff
@@ -1106,7 +1113,7 @@ static void
     safe_m = m;
     matrix_size = safe_m * safe_m * sizeof(@typ@);
     pivot_size = safe_m * sizeof(fortran_int);
-    tmp_buff = (npy_uint8 *)malloc(matrix_size + pivot_size);
+    tmp_buff = (npy_uint8 *)CKMALLOC(matrix_size + pivot_size);
```

Comment:
I think you just found a bug... - if the system is out of memory, numpy will just do no computation, without giving any diagnostic message.

Reply:
Also, I think `malloc(0)` is undefined behavior on all platforms, so we should fix the , probably by adding
```
if (m == 0) {
    return;
}
```

Reply:
Yeah, I would rather change those macros to include an `assert(num > 0)` then a conditional on a specific platform that happens to not work around standard limitations.

Reply:
For this case, doing this looks like it breaks various tests. As it currently stands:

```
a = np.zeros((0, 0))
res = linalg.det(a)
print(res)
```

This would be 1.0 currently. If we swap it to return early to avoid malloc(0), then that'll be 0.0. If we want to preserve the current behavior, then also adding in some default values could fix this, is there a preference one way or another?